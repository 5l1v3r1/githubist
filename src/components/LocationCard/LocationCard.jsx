// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import { Box } from '..';

import styles from './LocationCard.scss';

type Props = {
  rank?: number,
  name: string,
  slug: string,
  totalRepositories: number,
  totalDevelopers?: number,
  language?: {
    name: string,
    slug: string,
  },
};

const LocationCard = ({
  rank,
  name,
  slug,
  totalRepositories,
  totalDevelopers,
  language,
}: Props) => (
  <Box>
    <Link to={`/location/${slug}`} className={styles.name}>
      {rank ? `#${rank} ${name}` : name}
    </Link>

    <div className={styles.meta}>
      {totalDevelopers && (
        <Link to={`/location/${slug}/developers`} className={styles.metaItem}>
          {totalDevelopers.toLocaleString()} Geli≈ütirici
        </Link>
      )}
      <Link to={`/location/${slug}/repositories`} className={styles.metaItem}>
        {totalRepositories.toLocaleString()} Repo
      </Link>
      {language && (
        <Link to={`/language/${language.slug}`} className={cx(styles.metaItem, styles.language)}>
          {language.name}
        </Link>
      )}
    </div>
  </Box>
);

LocationCard.defaultProps = {
  rank: undefined,
  language: undefined,
  totalDevelopers: undefined,
};

export default LocationCard;
