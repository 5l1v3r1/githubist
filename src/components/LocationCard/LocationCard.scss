@import '../../theme/theme';

.locationCard {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: map-get($spacers, large);
  border: 1px solid $text-muted;
  background-color: $white;
  border-radius: px-to-rem(2px);
  box-shadow: 3px 3px 0 #d5d5d6;
}

.name {
  color: $black-color;
  font-weight: 600;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}

.meta {
  display: flex;
  flex-direction: row;
  margin: map-get($spacers, 2x-small) 0 0 0;
}

.metaItem {
  margin-right: map-get($spacers, normal);
  color: $body-color;
  font-size: map-get($font-sizes, small);
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}

.language {
  margin-right: 0;
  margin-left: auto;
  color: $secondary-color;
  font-weight: 600;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}
