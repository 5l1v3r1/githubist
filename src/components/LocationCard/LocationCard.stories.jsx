// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import backgrounds from '@storybook/addon-backgrounds';
import { MemoryRouter } from 'react-router-dom';

import LocationCard from './LocationCard';

storiesOf('LocationCard', module)
  .addDecorator(backgrounds([{ name: 'light', value: '#fcfafa', default: true }]))
  .add('default', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <LocationCard
          name="İzmir"
          slug="izmir"
          totalRepositories={2150}
          totalDevelopers={768}
          language={{ name: 'JavaScript', slug: 'javascript' }}
        />
      </div>
    </MemoryRouter>
  ))
  .add('with rank', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <LocationCard
          rank={1}
          name="İzmir"
          slug="izmir"
          totalRepositories={2150}
          totalDevelopers={768}
          language={{ name: 'JavaScript', slug: 'javascript' }}
        />
      </div>
    </MemoryRouter>
  ))
  .add('without language', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <LocationCard
          rank={1}
          name="İzmir"
          slug="izmir"
          totalRepositories={2150}
          totalDevelopers={768}
        />
      </div>
    </MemoryRouter>
  ));
