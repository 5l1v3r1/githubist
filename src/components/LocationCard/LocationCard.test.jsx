// @flow

import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import LocationCard from './LocationCard';

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LocationCard
        name="İzmir"
        slug="izmir"
        totalRepositories={2150}
        totalDevelopers={768}
        language={{ name: 'JavaScript', slug: 'javascript' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with rank', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LocationCard
        rank={1}
        name="İzmir"
        slug="izmir"
        totalRepositories={2150}
        totalDevelopers={768}
        language={{ name: 'JavaScript', slug: 'javascript' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without language', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LocationCard
        rank={1}
        name="İzmir"
        slug="izmir"
        totalRepositories={2150}
        totalDevelopers={768}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without developers count', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LocationCard
        rank={1}
        name="İzmir"
        slug="izmir"
        totalRepositories={2150}
        language={{ name: 'JavaScript', slug: 'javascript' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
