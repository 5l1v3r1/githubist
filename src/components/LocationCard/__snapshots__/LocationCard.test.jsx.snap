// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LocationCard
      language={
        Object {
          "name": "JavaScript",
          "slug": "javascript",
        }
      }
      name="İzmir"
      slug="izmir"
      totalDevelopers={768}
      totalRepositories={2150}
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <Link
            replace={false}
            to="/location/izmir"
          >
            <a
              href="/location/izmir"
              onClick={[Function]}
            >
              İzmir
            </a>
          </Link>
          <div>
            <Link
              replace={false}
              to="/location/izmir/developers"
            >
              <a
                href="/location/izmir/developers"
                onClick={[Function]}
              >
                768
                 Geliştirici
              </a>
            </Link>
            <Link
              replace={false}
              to="/location/izmir/repositories"
            >
              <a
                href="/location/izmir/repositories"
                onClick={[Function]}
              >
                2,150
                 Repo
              </a>
            </Link>
            <Link
              className=""
              replace={false}
              to="/language/javascript"
            >
              <a
                className=""
                href="/language/javascript"
                onClick={[Function]}
              >
                JavaScript
              </a>
            </Link>
          </div>
        </div>
      </Box>
    </LocationCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly with rank 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LocationCard
      language={
        Object {
          "name": "JavaScript",
          "slug": "javascript",
        }
      }
      name="İzmir"
      rank={1}
      slug="izmir"
      totalDevelopers={768}
      totalRepositories={2150}
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <Link
            replace={false}
            to="/location/izmir"
          >
            <a
              href="/location/izmir"
              onClick={[Function]}
            >
              #1 İzmir
            </a>
          </Link>
          <div>
            <Link
              replace={false}
              to="/location/izmir/developers"
            >
              <a
                href="/location/izmir/developers"
                onClick={[Function]}
              >
                768
                 Geliştirici
              </a>
            </Link>
            <Link
              replace={false}
              to="/location/izmir/repositories"
            >
              <a
                href="/location/izmir/repositories"
                onClick={[Function]}
              >
                2,150
                 Repo
              </a>
            </Link>
            <Link
              className=""
              replace={false}
              to="/language/javascript"
            >
              <a
                className=""
                href="/language/javascript"
                onClick={[Function]}
              >
                JavaScript
              </a>
            </Link>
          </div>
        </div>
      </Box>
    </LocationCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly without developers count 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LocationCard
      language={
        Object {
          "name": "JavaScript",
          "slug": "javascript",
        }
      }
      name="İzmir"
      rank={1}
      slug="izmir"
      totalRepositories={2150}
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <Link
            replace={false}
            to="/location/izmir"
          >
            <a
              href="/location/izmir"
              onClick={[Function]}
            >
              #1 İzmir
            </a>
          </Link>
          <div>
            <Link
              replace={false}
              to="/location/izmir/repositories"
            >
              <a
                href="/location/izmir/repositories"
                onClick={[Function]}
              >
                2,150
                 Repo
              </a>
            </Link>
            <Link
              className=""
              replace={false}
              to="/language/javascript"
            >
              <a
                className=""
                href="/language/javascript"
                onClick={[Function]}
              >
                JavaScript
              </a>
            </Link>
          </div>
        </div>
      </Box>
    </LocationCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly without language 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LocationCard
      name="İzmir"
      rank={1}
      slug="izmir"
      totalDevelopers={768}
      totalRepositories={2150}
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <Link
            replace={false}
            to="/location/izmir"
          >
            <a
              href="/location/izmir"
              onClick={[Function]}
            >
              #1 İzmir
            </a>
          </Link>
          <div>
            <Link
              replace={false}
              to="/location/izmir/developers"
            >
              <a
                href="/location/izmir/developers"
                onClick={[Function]}
              >
                768
                 Geliştirici
              </a>
            </Link>
            <Link
              replace={false}
              to="/location/izmir/repositories"
            >
              <a
                href="/location/izmir/repositories"
                onClick={[Function]}
              >
                2,150
                 Repo
              </a>
            </Link>
          </div>
        </div>
      </Box>
    </LocationCard>
  </Router>
</MemoryRouter>
`;
